x/gov/abci.go

```diff
@@ -72,6 +72,103 @@ func EndBlocker(ctx sdk.Context, keeper *keeper.Keeper) error {
 		return err
 	}
 
+	// fetch proposals that are due to be checked for quorum
+	rng = collections.NewPrefixUntilPairRange[time.Time, uint64](ctx.BlockTime())
+	err = keeper.QuorumCheckQueue.Walk(ctx, rng, func(key collections.Pair[time.Time, uint64], quorumCheckEntry v1.QuorumCheckQueueEntry) (bool, error) {
+		proposal, err := keeper.Proposals.Get(ctx, key.K2())
+		if err != nil {
+			return false, err
+		}
+
+		var tagValue, logMsg string
+
+		params, err := keeper.Params.Get(ctx)
+		if err != nil {
+			return false, err
+		}
+
+		// remove from queue
+		err = keeper.QuorumCheckQueue.Remove(ctx, key)
+		if err != nil {
+			return false, err
+		}
+
+		// check if proposal passed quorum
+		quorum, err := keeper.HasReachedQuorum(ctx, proposal)
+		if err != nil {
+			return false, err
+		}
+
+		logMsg = "proposal did not pass quorum after timeout, but was removed from quorum check queue"
+		tagValue = types.AttributeValueProposalQuorumNotMet
+
+		if quorum {
+			logMsg = "proposal passed quorum before timeout, vote period was not extended"
+			tagValue = types.AttributeValueProposalQuorumMet
+			if len(quorumCheckEntry.QuorumCheckTimestamps) == 0 {
+				// proposal passed quorum after timeout, extend voting period.
+				// canonically, we consider the first quorum check to be "right after" the  quorum timeout has elapsed,
+				// so if quorum is reached at the first check, we don't extend the voting period.
+				endTime := ctx.BlockTime().Add(*params.MaxVotingPeriodExtension)
+				logMsg = fmt.Sprintf("proposal passed quorum after timeout, but vote end %s is already after %s", proposal.VotingEndTime, endTime)
+				if endTime.After(*proposal.VotingEndTime) {
+					logMsg = fmt.Sprintf("proposal passed quorum after timeout, vote end was extended from %s to %s", proposal.VotingEndTime, endTime)
+					proposal.VotingEndTime = &endTime
+					err = keeper.SetProposal(ctx, proposal)
+					if err != nil {
+						return false, err
+					}
+				}
+			}
+		} else if uint64(len(quorumCheckEntry.QuorumCheckTimestamps)) < quorumCheckEntry.QuorumCheckCount && proposal.VotingEndTime.After(ctx.BlockTime()) {
+			// proposal did not pass quorum and is still active, add back to queue with updated time key and counter.
+			// compute time interval between quorum checks
+			quorumCheckPeriod := proposal.VotingEndTime.Sub(*quorumCheckEntry.QuorumTimeoutTime)
+			t := quorumCheckPeriod / time.Duration(params.QuorumCheckCount)
```

Comment:
```suggestion
			t := quorumCheckPeriod / time.Duration(quorumCheckEntry.QuorumCheckCount)
```

It's nice that you added this in the queue entry! You just forgot to use it here if I'm not wrong.